utility-emissions-channel/typescript_app/tests/test.js

```diff
@@ -64,62 +70,90 @@ describe("Test fabric", function() {
         orgName: "auditor1",
         utilityId: "USA_EIA_11208",
         partyId: "1234567890",
-        fromDate: "2020-04-06T10:10:09Z",
-        thruDate: "2020-04-06T10:10:09Z",
+        fromDate: "2018-05-07T10:10:09Z",
+        thruDate: "2018-05-07T10:10:09Z",
         energyUseAmount: 100,
         energyUseUom: "TONS",
       })
       .end((error, response) => {
         
-        let res = JSON.stringify(response.body.info.includes("Failed to submit transaction"));
-        if(res)
-        {
-        assert(response.status == 409, "Could not reach the gateway network");
+       try{
+        //console.log(`response :  ${JSON.stringify(response.body)}`);
+        expect(response.body.info).to.equal("EMISSION RECORDED TO LEDGER");
         done();
-        }
-        
-        else if(response.body.info == "EMISSION RECORDED TO LEDGER")
-        {
-        assert(response.body.info == "EMISSION RECORDED TO LEDGER", "Emission is successfully recorded.");
-        done();
-        }
+       }
         
-        else if(error) 
+       catch (error)
         {
           done(error);
         } 
        
-       else{
-          done();
-          }
-        
-        
       });
   });
  
  
-  it("should display emission records for the correct year", function(done) {
+  it("should display emission records for the correct year (2019)", function(done) {
     chai
        .request(host)
-       .get(getAllEmissionsPath)
-       .set("content-type", "application/json")
-       .end((error, response) => {
-           
-        if(response.body == [])
-        {
-          assert(response.body == [], "No emissions to display");
+      .post(recordEmissionPath)
+      // .field('myparam' , 'test')
+      .set("content-type", "application/x-www-form-urlencoded")
+      .send({
+        userId: "TestUser",
+        orgName: "auditor1",
+        utilityId: "USA_EIA_11208",
+        partyId: "1234567890",
+        fromDate: "2019-04-06T10:10:09Z",
+        thruDate: "2019-04-06T10:10:09Z",
+        energyUseAmount: 100,
+        energyUseUom: "TONS",
+      })
+      .end((error, response) => {
+       
+       try{
+          
+          let entry = response.body;
+          expect(entry.info).to.equal("EMISSION RECORDED TO LEDGER");
+          let year = entry.factorSource.includes("2019");
+          expect(year).to.equal(true);
           done();
-        }
-        
-        else if(response.body[0])
+       }
+
+        catch(error) 
         {
-          let entry = response.body[0];
+          done(error);
+        } 
+      });
+  });
+  
```

Comment:
Hi @sichen1234 ,

Did you mean this ?
I am first recording the emission for year 2019, and then checking if the response has the emission recorded for year 2019.